### PR TITLE
BAU - Add mongock as a DB migration tool

### PR DESCRIPTION
**Pull request by PauloPortugal**
https://github.com/cloudyrock/mongock is an updated fork from Mongobee

### CODE REVIEW

File: project/AppDependencies.scala [hunk 1]
```diff
@@ -8,15 +8,15 @@ object AppDependencies {
...
-    "com.github.tototoshi"  %% "scala-csv"                  % "1.3.6",
-    "uk.gov.hmrc"           %% "play-json-union-formatter"  % "1.7.0",
-    "com.github.mongobee"   %  "mongobee"                   % "0.13"
+    "uk.gov.hmrc"                     %% "bootstrap-play-26"          % "1.3.0",
+    "uk.gov.hmrc"                     %% "wco-dec"                    % "0.33.0",
+    "uk.gov.hmrc"                     %% "logback-json-logger"        % "4.6.0",
+    "com.typesafe.play"               %% "play-json-joda"             % "2.6.13",
+    "com.github.tototoshi"            %% "scala-csv"                  % "1.3.6",
+    "uk.gov.hmrc"                     %% "play-json-union-formatter"  % "1.7.0",
+    "com.github.cloudyrock.mongock"   %  "mongock-core"               % "2.0.2"
```
Comment (PatrykRudnicki) [inline project/AppDependencies.scala]: Do you need add:
`org.mongodb" %  "mongo-java-driver" % "3.12.1`

Comment (PauloPortugal) [reply project/AppDependencies.scala]: Yes and we also need to override.
Although it worked correctly, but you are spot on.

File: build.sbt [hunk 2]
```diff
@@ -36,6 +36,7 @@ lazy val microservice = Project(appName, file("."))
   .settings(
     libraryDependencies ++= (AppDependencies.compile ++ AppDependencies.test()).map(_.withSources),
     dependencyOverrides ++= AppDependencies.jettyOverrides,
+    dependencyOverrides += "org.mongodb" % "mongo-java-driver" % "3.12.1",
```
Comment (PatrykRudnicki) [inline build.sbt]: Are you sure that overriding is necessary? `customs-declare-exports` doesn't have it. Should I add this to service?

Comment (PauloPortugal) [reply build.sbt]: It supports all MongoDB versions, so this avoids us having problems in the near and medium term
https://docs.mongodb.com/ecosystem/drivers/java/

**Pull request merged**